lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -306,7 +306,7 @@ void LCIStrategicPlugin::handleCruisingUntilStop(const cav_srvs::PlanManeuversRe
   std::vector<lanelet::ConstLanelet> lane_follow_crossed_lanelets =
       getLaneletsBetweenWithException(new_ts_params.x1_, new_ts_params.x2_, true, true);
 
-  resp.new_plan.maneuvers.push_back(composeTrajectorySmoothingManeuverMessage(current_state.downtrack, traffic_light_down_track, lane_follow_crossed_lanelets, 
+  resp.new_plan.maneuvers.push_back(composeTrajectorySmoothingManeuverMessage(current_state.downtrack, new_ts_params.x2_, lane_follow_crossed_lanelets, 
```

Comment:
What is x2 here? And how is that different from traffic_light_down_track?

Reply:
CASE8 means "cruise until stop", so the algorithm generates cruising from x0 to x1 to x2, and stopping maneuver from x2 to x3 (which is the stop_line or traffic_light_down_track). So TrajectorySmoothingManeuver (lci_tactical) should be only until x2, and StopAndWaitManeuever should be from x2 to x3 (traffic_light_downtrack)